### ISSUE TITLE
[strict-boolean-expressions] Support !! "operator" as alternative to Boolean() cast

### ISSUE THREAD

**Issue by Jessidhia** [typescript-eslint/typescript-eslint#3470]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/strict-boolean-expressions": "error"
  }
}
```

```TS
function x(str: string | false) {
  if (!!str) {
    return true
  }
  return false
}
```

**Expected Result**

`!!` is a common expression used to enforce that a value is treated as a boolean, so it should be accepted as equivalent to wrapping in the `Boolean()` constructor-function.

**Actual Result**

```
Unexpected value in conditional. A boolean expression is required.
```

**Additional Info**

Either `Boolean()` or `!!` tend to trigger eslint's builtin `no-extra-boolean-cast` so it might be worth also mentioning it in the rule's manual.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.13.0` |
| `@typescript-eslint/parser`        | `4.13.0` |
| `TypeScript`                       | `4.1.2` |
| `ESLint`                           | `7.15.0` |
| `node`                             | `14.15.4` |

**Comment by bradzacher** [comment]
The point of the rule is that you explicitly handle each case when multiple false cases exist.

For example, say your argument started as type `string` and you use `!!str`. Then say you change the type of the argument to `string | false`. Now what? Is `!!str` still the correct code? Is it intentional that you're handling the empty string and `false` cases the same? Who knows - it's not clear in the code what the intention exactly is.

The same applies to `Boolean`.

Instead of relying on implicit coercion - instead if you explicitly handle each case you end up with clear, self documenting code.

```ts
function​ ​x​(​str​: ​string​ ​|​ ​false​)​ ​{​
  ​if​ ​(​str !== false && str​ !== '')​ ​{​
    ​return​ ​true​
  ​}​
  ​return​ ​false​
​}
```

Code is read more times than it's written - always optimise for readability and clarity over terseness.
